pykickstart/parser.py

```diff
@@ -193,10 +193,12 @@ def __str__(self):
             retval += '\n%post'
         elif self.type == constants.KS_SCRIPT_TRACEBACK:
             retval += '\n%traceback'
+        elif self.type == constants.KS_SCRIPT_PREINSTALL:
+            retval += '\n%pre-install'
 
         if self.interp != "/bin/sh" and self.interp != "":
             retval += " --interpreter=%s" % self.interp
-        if self.type == constants.KS_SCRIPT_POST and not self.inChroot:
+        if self.type in [constants.KS_SCRIPT_POST, constants.KS_SCRIPT_PREINSTALL] and not self.inChroot:
```

Comment:
This seems to be not aligned with ``Like %pre these scripts do not run in the chrooted environment.``.